src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/DesignTimeTagHelperWriter.cs

```diff
@@ -32,26 +38,23 @@ public override void WriteAddTagHelperHtmlAttribute(CSharpRenderingContext conte
             context.RenderChildren(node);
         }
 
-        public override void WriteCreateTagHelper(CSharpRenderingContext context, CreateTagHelperIRNode node)
-        {
-            var tagHelperVariableName = GetTagHelperVariableName(node.TagHelperTypeName);
-
-            context.Writer
-                .WriteStartAssignment(tagHelperVariableName)
-                .WriteStartMethodInvocation(
-                    CreateTagHelperMethodName,
-                    "global::" + node.TagHelperTypeName)
-                .WriteEndMethodInvocation();
-        }
-
-        public override void WriteExecuteTagHelpers(CSharpRenderingContext context, ExecuteTagHelpersIRNode node)
-        {
-            // Do nothing
-        }
-
         public override void WriteInitializeTagHelperStructure(CSharpRenderingContext context, InitializeTagHelperStructureIRNode node)
         {
             context.RenderChildren(node);
+
+            // CreateTagHelper
+            foreach (var descriptor in node.TagHelperBinding.Descriptors)
```

Comment:
Agree that what you have to do isn't beautiful but it cleans up the IR surface area. Note: You could always just do `bodyNode.Children.Skip(1)` 👍 

By having all the info on `TagHelperIRNode` you can comfortably rename this node from initialize structure to `TagHelperBodyIRNode`